packages/drivers/odsp-driver/src/fetchSnapshot.ts

```diff
@@ -51,3 +62,235 @@ export async function fetchSnapshot(
         async () => epochTracker.fetchAndParseAsJSON<IOdspSnapshot>(url, { headers }, "snapshotTree"),
     );
 }
+
+export async function fetchLatestSnapshotCore(
+    odspResolvedUrl: IOdspResolvedUrl,
+    storageTokenFetcher: (options: TokenFetchOptions, name: string) => Promise<string | null>,
+    tokenFetchOptions: TokenFetchOptions,
+    snapshotOptions: ISnapshotOptions | undefined,
+    logger: ITelemetryLogger,
+    summarizerClient: boolean = false,
+    epochTracker: EpochTracker | undefined,
+    snapshotCacheEntry: IEntry,
+    persistedCache: IPersistedCache,
+): Promise<{
+    odspSnapshot: ISnapshotCacheValue,
+    cacheP?: Promise<void>,
+}> {
+    const snapshotUrl = odspResolvedUrl.endpoints.snapshotStorageUrl;
+    const url = `${snapshotUrl}/trees/latest?ump=1`;
+    const storageToken = await storageTokenFetcher(tokenFetchOptions, "TreesLatest");
+    assert(storageToken !== null, "Storage token should not be null");
+    const formBoundary = uuid();
+    const formParams: string[] = [];
+    formParams.push(`--${formBoundary}`);
+    formParams.push(`Authorization: Bearer ${storageToken}`);
+    formParams.push(`X-HTTP-Method-Override: GET`);
+    const logOptions = {};
+    if (snapshotOptions !== undefined) {
+        Object.entries(snapshotOptions).forEach(([key, value]) => {
+            if (value !== undefined) {
+                formParams.push(`${key}: ${value}`);
+                logOptions[`snapshotOption_${key}`] = value;
+            }
+        });
+    }
+    if (odspResolvedUrl.sharingLinkToRedeem) {
+        formParams.push(`sl: ${odspResolvedUrl.sharingLinkToRedeem}`);
+    }
+    formParams.push(`_post: 1`);
+    formParams.push(`\r\n--${formBoundary}--`);
+    const postBody = formParams.join("\r\n");
+    const headers: {[index: string]: any} = {
+        "Content-Type": `multipart/form-data;boundary=${formBoundary}`,
+    };
+
+    let controller: AbortController | undefined;
+    if (summarizerClient !== true && snapshotOptions?.timeout !== undefined) {
+        controller = new AbortController();
+        setTimeout(
+            () => controller!.abort(),
+            snapshotOptions.timeout,
+        );
+    }
+
+    // This event measures only successful cases of getLatest call (no tokens, no retries).
+    return PerformanceEvent.timedExecAsync(
+        logger,
+        {
+            eventName: "TreesLatest",
+            ...logOptions,
+        },
+        async (event) => {
+            const startTime = performance.now();
+            let response: IOdspResponse<IOdspSnapshot>;
+            if (epochTracker !== undefined) {
+                response = await epochTracker.fetchAndParseAsJSON<IOdspSnapshot>(
```

Comment:
If you can re-arrange caching logic below not to rely on epochTracker, then I'd suggest removing depends on epochTracker here as well, and instead let caller provide you a callback that is used to fetch snapshot.
It will be either fetchAndParseAsJSONHelper<IOdspSnapshot>, or epochTracker.fetchAndParseAsJSON<IOdspSnapshot>.bind(epochTracker)

Reply:
Removed dep on epoch tracker.